framer.go

```diff
@@ -140,3 +142,27 @@ func (f *framerI) AppendStreamFrames(frames []ackhandler.Frame, maxLen protocol.
 	}
 	return frames, length
 }
+
+func (f *framerI) Handle0RTTRejection() {
+	f.mutex.Lock()
+	f.controlFrameMutex.Lock()
+	f.streamQueue = f.streamQueue[:0]
+	for id := range f.activeStreams {
+		delete(f.activeStreams, id)
+	}
+	var j int
+	for i, frame := range f.controlFrames {
+		switch frame.(type) {
+		case *wire.MaxDataFrame, *wire.MaxStreamDataFrame, *wire.MaxStreamsFrame:
+			panic("didn't expect this frame to be sent in 0-RTT")
```

Comment:
Return an error? Seems like a rather complex condition for a server-crashing assert to me :)

(Then also defer the unlocks.)